src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java

```diff
@@ -1061,7 +1084,15 @@ public synchronized void handle(Message<T> event) {
         }
       } else {
         checkNextTick();
-        handler.handle(event);
+        MessageImpl abc = (MessageImpl) event;
+        metrics.beginHandleMessage(metric, abc.getSocket() == null);
+        try {
+          handler.handle(event);
+          metrics.endHandleMessage(metric, null);
```

Comment:
I still question its value. If the execution time is always approx. zero, what's the point?

I'd also be concerned about such fine grained metrics killing performance. Is there some way of enabling metrics but disabling the more fine grained stuff, or is metrics being enabled all or nothing?